packages/blade/src/utils/useBreakpoint/useBreakpoint.web.test.ts

```diff
@@ -0,0 +1,48 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+import breakpoints from '../../tokens/global/breakpoints';
+import setupMatchMediaMock from '../mocks/setupMatchMediaMock';
+import useBreakpoint from './';
+
+describe('useBreakpoint', () => {
+  it('should detect mobile and return smallest breakpoint of mobile devices', () => {
+    setupMatchMediaMock({
+      matches: true,
+    });
+
+    const { result } = renderHook(() => useBreakpoint({ breakpoints }));
+
+    expect(result.current.matchedBreakpoint).toBe('xs');
+    expect(result.current.matchedDeviceType).toBe('mobile');
+  });
+
+  it('should match a specific media query and return the device type accordingly', () => {
+    const addEventListenerMock = jest
+      .fn()
+      .mockImplementation((_, callback: (arg: { media: string }) => void) => {
```

Comment:
the reason why we do this is because we don't have a browser window here so all we do is mock the event listener and call it with a mediaQuery that we might receive in actual browser i.e `media: 'screen and (min-width: 769px) and (max-width: 1024px)'` and then do our assertions. 

Then after our assertions, we again call this callback explicitly [here](https://github.com/razorpay/blade/pull/424/files#diff-b9d80003bb947c85eab50a87209022f7751656022bef098c6655d561d488a5edR36) to simulate the media query change event and this time we pass different media query i.e `media: 'screen and (max-width: 320px)'` and then make assertions that we get different device as intended